### DISCUSSION TITLE
how to use APIRouter with pages as classes

### DISCUSSION THREAD

**Discussion by belalhmedan90** [zauberzeug/nicegui#3979] [Q&A]
### Question

I created pages as classes:
```py
from nicegui import ui

class ProgramPage:
    def __init__(self, program_name:str) -> None:
    self._program_name = program_name
        self.create_page()

    @ui.page('/smart_palletizing_program')
    def create_page(self):
        ui.label('My App')
        ui.run(show=False, title=f"{self._program_name}")

class IntroPage:
    def __init__(self):
        self.create_page()

    @ui.page('/smart_palletizing_program')
    def create_page(self):
        ui.label('My Intro')
        ui.run(show=False, title="Intro")
```
each of which of these classes is in different script, and I want to use `APIRouter` to create page paths, and route between them using main script e.g.: `app.py` can you please tell me how I can do that? otherwise can you please guide me to a best-practice repo that uses the Router with multi pages / classes? thanks

**Comment by rodja** [comment]
Have you seen our [modularization example](https://github.com/zauberzeug/nicegui/tree/main/examples/modularization/)? It shows some techniques on how to split you pages into several files and also includes an `APIRouter` demo.

**Reply by belalhmedan90** [reply]
@rodja Hi, and thanks for your kind answer, I have seen that, but I was wondering if there is an example of a router with multiple sub paths passed to classes to derive `main_path/sub_path (kind of more advanced example)`, thanks in advance for your time and valuable response.
